mediorum/cmd/reaper/reaper.go

```diff
@@ -0,0 +1,391 @@
+		counter[fileType] = make(map[string]int)
+		counter[fileType]["count"] = 0
+		counter[fileType]["error_count"] = 0
+		counter[fileType]["bytes_used"] = 0
+	}
+
+	return counter
+}
+
+func init() {
+	BATCH_SIZE = 1000
+
+	dbUrl = os.Getenv("dbUrl")
+	if dbUrl == "" {
+		dbUrl = "postgres://postgres:example@localhost:5454/m1"
+	}
+}
+
+func _init() {
+	reaperCmd = flag.NewFlagSet("reaper", flag.ExitOnError)
+	deleteFiles := reaperCmd.Bool("delete", false, "delete files (default false)")
+	logDir := reaperCmd.String("logDir", "/tmp/reaper/logs", "directory to store job logs (default: /tmp/reaper/logs)")
+	walkDir := reaperCmd.String("walkDir", "/tmp/reaper/to_walk", "directory to walk (default: /tmp/reaper/to_walk)")
+	reaperCmd.Parse(os.Args[2:])
+
+	config = &Config{
+		DeleteFiles: *deleteFiles,
+		WalkDir:     *walkDir,
+		LogDir:      *logDir,
+		dbUrl:       dbUrl,
+	}
+
+	fmt.Printf("config: %+v\n", config)
+}
+
+func _trap() {
+	// Catch control+c so you can check things are working on a large fs without waiting
+	interrupt := make(chan os.Signal, 1)
+	signal.Notify(interrupt, os.Interrupt, syscall.SIGTERM)
+
+	go func() {
+		<-interrupt
+		fmt.Printf("\ntrapped SIGINT...\n\n")
+		r.report()
+		os.Exit(0)
+	}()
+}
+
+func Run() {
+
+	_init() // Flag parsing in init() causes `go test` to fail
+
+	_trap()
+
+	var err error
+
+	r, err = NewReaper(config)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer r.Close()
+
+	err = r.Walk()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	r.report()
+}
+
+func (r *Reaper) Walk() error {
+
+	r.counter = initCounter()
+	r.Iteration = 0
+
+	err := r.walk(config.WalkDir)
+
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Process any remaining paths
+	if len(r.Batch) > 0 {
+		err = r.processBatch()
+		if err != nil {
+			fmt.Println("Error processing final batch:", err)
+			return err
+		}
+	}
+
+	if config.DeleteFiles {
+		r.dropSegmentRows()
+	}
+
+	return nil
+}
+
+func (r *Reaper) walk(path string) error {
+	entries, err := os.ReadDir(path)
+	if err != nil {
+		return fmt.Errorf("failed to read dir: %v", err)
+	}
+
+	for _, entry := range entries {
+		filePath := filepath.Join(path, entry.Name())
+
+		if entry.IsDir() {
+			err = r.walk(filePath)
+			if err != nil {
+				fmt.Printf("error walking path %q: %v\n", filePath, err)
+				continue
+			}
+		} else {
+			// Be kind to disk i/o
+			// time.Sleep(time.Millisecond * 2)
+			r.Batch = append(r.Batch, filePath)
+
+			if len(r.Batch) == BATCH_SIZE {
+				err = r.processBatch()
+				if err != nil {
+					fmt.Println("Error processing r.Batch:", err)
+					return err
+				}
+				r.Batch = r.Batch[:0] // Reset
+			}
+		}
+	}
+	return nil
+}
+
+func (r *Reaper) processBatch() error {
+
+	r.Iteration++
+	fmt.Println(fmt.Sprintf("Process batch: %d", r.Iteration))
+
+	tableName := "Files"
+	if r.Config.isTest {
+		tableName = "FilesTest"
+	}
+	query := fmt.Sprintf(`SELECT DISTINCT "storagePath", "type" FROM "%s" WHERE "storagePath" = ANY($1)`, tableName)
```

Comment:
this part is tricky because there are files that we still serve that don't have the right storagePath in the db. those possible paths are enumerated [here](https://github.com/AudiusProject/audius-protocol/blob/main/mediorum/server/serve_legacy.go#L209)

Reply:
we'd also want to exclude `google-application-credentials.json`